### PR TITLE
Kokoro py (WIP)

### PR DESCRIPTION
**Pull request by dlj-NaN**

### CODE REVIEW

File: kokoro/linux/python/resolve_docker_tag.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/python
+
+"""Retrieve an image's information."""
+
+import re
+import sys
+
+def _parse_image(name):
+    m = re.match(r'^(?:(?P<host>[\w.-^_]+(?:[:][\d]+))/)?(?P<name>[\w.-]+/[\w.-]+)(?:[:](?P<tag>[\w.-]+))?$',
```
Comment (fowles) [inline kokoro/linux/python/resolve_docker_tag.py:9]: can you add a comment with a few examples of strings you expect this to match against

File: kokoro/linux/python/x86_64/build_python.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#   "cpp" (default, C++ native extension) or "python" (pure Python)
+: ${PROTOBUF_PYTHON_RUNTIME:=cpp}
+
+# Validate the runtime selection.
+if [[ ${PROTOBUF_PYTHON_RUNTIME} == "cpp" ]]; then
+  BUILD_NATIVE=1
+elif [[ ${PROTOBUF_PYTHON_RUNTIME} != "python" ]]; then
+  echo -e "Unrecognized PROTOBUF_PYTHON_RUNTIME: ${PROTOBUF_PYTHON_RUNTIME@Q}" >&2
+  exit 1
+fi
```
Comment (fowles) [inline kokoro/linux/python/x86_64/build_python.sh:20]: shell scripts should almost always run with set -ex

File: kokoro/linux/python/x86_64/build_python.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  (
+    cd python
+    python setup.py build
+    pip install -e .
+  )
+}
+
+# Builds the native extension runtime.
+build_runtime_cpp() {
+  (
```
Comment (fowles) [inline kokoro/linux/python/x86_64/build_python.sh:42]: why the parens here?

File: kokoro/linux/python/x86_64/kokoro_build.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# The following values can be overridden by environment variable:
+#   DOCKER_IMAGE
+#   PROTOBUF_WORKSPACE
+#   PROTOBUF_PYTHON_RUNTIME (used in build_python.sh)
+
+# The Docker image to run the build.
+#
+# The official Python images are based on buildpack-deps, which is
+# sufficient to build the Protobuf project. The versioned builds specify
+# the image via environment variable, but this is a reasonable default:
```
Comment (fowles) [inline kokoro/linux/python/x86_64/kokoro_build.sh:16]: set ex

File: kokoro/linux/python/x86_64/build_python.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  BUILD_NATIVE=1
+elif [[ ${PROTOBUF_PYTHON_RUNTIME} != "python" ]]; then
+  echo -e "Unrecognized PROTOBUF_PYTHON_RUNTIME: ${PROTOBUF_PYTHON_RUNTIME@Q}" >&2
+  exit 1
+fi
+
+# Builds the protobuf library. At a minimum, this is needed for protoc.
+build_libprotobuf() {
+  # To build the C++ extension, we must make sure to build PIC.
+  ./autogen.sh
```
Comment (haberman) [inline kokoro/linux/python/x86_64/build_python.sh:25]: It is possible to use Bazel instead? I've been trying to reduce dependencies on Automake in hopes we can kill it eventually.

File: kokoro/linux/python/x86_64/kokoro_build.sh [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+: ${DOCKER_IMAGE:=python:buster}
+
+# The location for the source checkout.
```
Comment (haberman) [inline kokoro/linux/python/x86_64/kokoro_build.sh:19]: Can this support interactive invocation from a working directory, for easy testing during development? I recently did this for our PHP images:

https://github.com/protocolbuffers/protobuf/blob/master/kokoro/linux/php_all/build.sh

**Comment by acozzette** [comment]
@dlj-NaN I hope you don't mind if I close this old pull request. Feel free to reopen it if it's still relevant of course.

**Pull request closed without merging**